soundwire-defconfig

```diff
@@ -1,16 +1,27 @@
 CONFIG_SOUNDWIRE=m
 CONFIG_SOUNDWIRE_INTEL=m
-CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE_LINK=y
+CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE=m
 
 # add codecs so that they are compiled even if no machine
 # drivers are selected
-CONFIG_SND_SOC_RT700_SDW=m
-CONFIG_SND_SOC_RT711_SDW=m
+CONFIG_MFD_CS42L43_SDW=m
+CONFIG_SND_SOC_CS35L56_SDW=m
+CONFIG_SND_SOC_CS42L42_SDW=m
+CONFIG_SND_SOC_CS42L43_SDW=m
+CONFIG_SND_SOC_MAX98373_SDW=m
+
+CONFIG_SND_SOC_RT1017_SDCA_SDW=m
 CONFIG_SND_SOC_RT1308_SDW=m
-CONFIG_SND_SOC_RT715_SDW=m
-CONFIG_SND_SOC_RT711_SDCA_SDW=m
 CONFIG_SND_SOC_RT1316_SDW=m
 CONFIG_SND_SOC_RT1318_SDW=m
+CONFIG_SND_SOC_RT5682_SDW=m
+CONFIG_SND_SOC_RT700_SDW=m
+CONFIG_SND_SOC_RT711_SDW=m
+CONFIG_SND_SOC_RT711_SDCA_SDW=m
+CONFIG_SND_SOC_RT712_SDCA_SDW=m
+CONFIG_SND_SOC_RT712_SDCA_DMIC_SDW=m
+CONFIG_SND_SOC_RT722_SDCA_SDW=m
+CONFIG_SND_SOC_RT715_SDW=m
```

Comment:
Hmm, I am thinking should codec driver be selected by machine drivers? And we will see the issue if a machine driver doesn't select required codec driver in Kconfig.

Reply:
it's a good question. 

The problem is that for SoundWire maintenance we want to compile all possible codecs. If we don't we might have compilation misses when we change SoundWire APIs, etc.
But on the other hand, as you mentioned if we use them all then we can't figure out missing dependencies for boards.

I guess I am biased on the first option...

Reply:
Why not both? Using two different subsets of `defconfig` files. Github workflows are great for this sort of compilation test.

Reply:
Not a bad idea @marc-hb, this really depends on whether you are a developer or not, so we could use soundwire-defconfig for the sof-defconfig and soundwire-dev-defconfig for the sof-dev-defconfig. Intel CI uses the latter, so as you said we could have a GitHub action for the minimal sof-defconfig stuff for non-developers.

Reply:
and updating the machine driver every time when a new SDW codec is added? That does not make much sense, right?

Reply:
> and updating the machine driver every time when a new SDW codec is added? That does not make much sense, right?

If we added a new codec support to a machine driver, I think it makes sense that we need to select the codec driver if the machine driver is selected.
Like
```
config SND_SOC_INTEL_SOUNDWIRE_SOF_MACH
        tristate "SoundWire generic machine driver"
        depends on I2C && ACPI
        depends on MFD_INTEL_LPSS || COMPILE_TEST
        depends on SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES || COMPILE_TEST
        depends on SOUNDWIRE
        select SND_SOC_MAX98363
        select SND_SOC_MAX98373_I2C
        select SND_SOC_MAX98373_SDW
        select SND_SOC_RT700_SDW
        select SND_SOC_RT711_SDW
        select SND_SOC_RT711_SDCA_SDW
        select SND_SOC_RT712_SDCA_SDW
        select SND_SOC_RT712_SDCA_DMIC_SDW
        select SND_SOC_RT715_SDW
        select SND_SOC_RT715_SDCA_SDW
        select SND_SOC_RT722_SDCA_SDW
        select SND_SOC_RT1308_SDW
        select SND_SOC_RT1308
        select SND_SOC_RT1316_SDW
        select SND_SOC_RT1318_SDW
        select SND_SOC_RT5682_SDW
        select SND_SOC_CS42L42_SDW
        select SND_SOC_CS42L43
        select SND_SOC_CS42L43_SDW
        select MFD_CS42L43
        select MFD_CS42L43_SDW
        select SND_SOC_CS35L56_SDW
        select SND_SOC_DMIC
        select SND_SOC_INTEL_HDA_DSP_COMMON
        select SND_SOC_INTEL_SOF_MAXIM_COMMON
        imply SND_SOC_SDW_MOCKUP
```

Reply:
@ujfalusi the sof-sdw machine driver pretty much supports all existing SoundWire devices, so we have to select them all to avoid missing dependencies on dais.